internal/server/index.go

```diff
@@ -15,21 +17,39 @@ import (
 var alphaNumericRunes = []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789")
 
 // ServeIndex serve the index.html file with nonce generated for supporting
-// restrictive CSP while using material-ui.
+// restrictive CSP while using material-ui from the local or embedded
+// virtual filesystem.
 func ServeIndex(publicDir string) fasthttp.RequestHandler {
-	f, err := os.Open(publicDir + "/index.html")
+	var f io.Reader
+	var err error
+
+	if publicDir == "/public_html" {
+		f, err = br.Open(publicDir + "/index.html")
+	} else {
+		f, err = os.Open(publicDir + "/index.html")
+	}
+
 	if err != nil {
 		logging.Logger().Fatalf("Unable to open index.html: %v", err)
+		return func(ctx *fasthttp.RequestCtx) {
```

Comment:
Sorry, now that I see it written it looks weird to me. I think it would be better to make the `ServeIndex` function return (fasthttp.RequestHandler, error) and panic in the caller in case of error. Indeed, Authelia should not even start in those cases since this code is only executed at startup.

Reply:
I've just realised neither of these should be required, `logging.Logger().Fatalf` will obviously exit gracefully once the error has been logged.

Any issues with the embedded filesystem are handled and you handle executing the template by returning a 503, perhaps instead we actually change this to a 500?

Reply:
Ah yes you're right, Fatalf does the job. For 500 vs 503, just align with the other handlers.